### PR TITLE
fix(components): [el-input-number]do not change undefined to NaN

### LINKED ISSUE #4996: [Bug Report] el-input-number 如果赋默认值 undefined 保存的时候会被置为 NaN,导致后端保存异常！！！Invalid prop: type check failed for prop "modelValue". Expected Number with value NaN
### Element Plus version
1.2.0-beta.6

### OS/Browsers version
12.0.1/360极速浏览器12.2.1650.0

### Vue version
3.2.26

### Reproduction Link
https://elementui.github.io/issue-generator/#/en-US?repo=element-plus

### Steps to reproduce
1.使用el-input-number;
2.初始值置为 undefined；
3.保存的时候值为 NaN;

### What is Expected?
如果是 undefined，不进行转换为 NaN

### What is actually happening?
NaN后端保存报错

### PR DESCRIPTION
**Pull request by Alanscut**
fix #4996 
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/input-number/__tests__/input-number.spec.ts [hunk 1]
```diff
@@ -41,6 +41,22 @@ describe('InputNumber.vue', () => {
...
     expect(wrapper.find('input').element.value).toEqual('1')
   })
+  test('set modelValue undefined to form validate', async () => {
+    const wrapper = _mount({
+      template:
+        '<el-input-number :model-value="formData.num" placeholder="input number"/><p>{{formData.num}}</p>',
+      setup() {
+        const formData = ref({
+          num: undefined,
+        })
```
Comment (JeremyWuuuuu) [inline packages/components/input-number/__tests__/input-number.spec.ts:51]: I thin we can just use 

```ts
const num = ref(undefined)
```

**Review by jw-foss — COMMENTED** [review]
Left a comment for you to consider, let me know if you have any questions

File: packages/components/input-number/__tests__/input-number.spec.ts (hunk 1 above)
Comment (Alanscut) [reply packages/components/input-number/__tests__/input-number.spec.ts:51]: I think it makes no difference, if you think it needs to be modified, I can modify it

**Comment by Alanscut** [comment]
@JeremyWuuuuu 
it has been changed, please review again

**Comment by CarterLi** [comment]
I'm encountering this issue, any chance gets merged?

**Comment by CarterLi** [comment]
@Alanscut 建议把 null 值也处理一下。因为后端java是没有undefined的，通过json返回的数据都是null。目前情况null值会被转化为0（或minValue）。所有可空字段都有这样的问题

**Comment by Alanscut** [comment]
> 

ok

**Comment by Alanscut** [comment]
我个人认为值为null时确实需要显示`placeholder`文本而不是被自动改为0.

**Comment by CarterLi** [comment]
@Alanscut 现在的逻辑，null会变成NaN？

**Comment by Alanscut** [comment]
> @Alanscut 现在的逻辑，null会变成NaN？

没有，变的是组件内部的` data.currentValue = newVal`，props.modelvalue没有被改变

https://user-images.githubusercontent.com/21104054/149612406-290db4d2-2352-48ff-b260-52d06917876e.mp4

image

**Review by jw-foss — APPROVED** [review]

**Pull request merged**